sdk/storage/azure-storage-blob/src/main/java/com/azure/storage/blob/BlobContainerAsyncClient.java

```diff
@@ -980,8 +982,17 @@ public String getEncryptionScope() {
      */
     PagedFlux<BlobItem> listBlobsHierarchyWithOptionalTimeout(String delimiter, ListBlobsOptions options,
         Duration timeout) {
-        Function<String, Mono<PagedResponse<BlobItem>>> func =
-            marker -> listBlobsHierarchySegment(marker, delimiter, options, timeout)
+        BiFunction<String, Integer, Mono<PagedResponse<BlobItem>>> func =
+            (marker, pageSize) -> {
+                ListBlobsOptions finalOptions = null;
+                if (pageSize != null) {
```

Comment:
What happens if a user sets the ListBlobsOptions maxResultsPerPage and the iterableByPage(size)? Should we deprecate our way of doing it?

Reply:
Discussed offline, but I think keeping both is fine because this is specific to the sync case. If the customer sets a value using iterableByPage, it'll overwrite the one from the options, but if they call iterableByPage() with no arg, it'll fall back to the options

Reply:
Should we put this in docs?